shell/app/node_main.cc

```diff
@@ -167,13 +146,11 @@ int NodeMain(int argc, char* argv[]) {
       // TODO(codebytere): we shouldn't have to call this - upstream?
       env->InitializeDiagnostics();
 
-      // This is needed in order to enable v8 host weakref hooks.
-      // TODO(codebytere): we shouldn't have to call this - upstream?
-      isolate->SetHostCleanupFinalizationGroupCallback(
-          HostCleanupFinalizationGroupCallback);
+      // We do not want to set the error level.
+      node::IsolateSettings is;
+      is.flags = ~node::IsolateSettingsFlags::MESSAGE_LISTENER_WITH_ERROR_LEVEL;
```

Comment:
Why don't we want to set this ? what happens if we do it ?

Reply:
Electron relies on error reporting matching that of Blink/Chromium, which Node.js changes. If we set this it kills ~40 previously-passing tests (see [here](https://app.circleci.com/pipelines/github/electron/electron/23480/workflows/98efc491-72d9-46fe-9310-ff737775a34c/jobs/528118/tests)) since Electron is more sensitive to errors and will exit the process instead of deferring to other listeners in place.

Reply:
> Electron relies on error reporting matching that of Blink/Chromium.
> .., since Electron is more sensitive to errors and will exit the process instead of deferring to other listeners in place.

Can you expand on how this behavior is inherited in `ELECTRON_RUN_AS_NODE` mode ?

I would expect the node env thats run in the child process to not have interference from the chromium process management.

Reply:
Sorry, 'relies' is not the best term. It collides with changes we've already made in [this patch](https://github.com/electron/electron/blob/fdf7e288bb049a681a46351e7770b68f41b5075e/patches/node/feat_add_flags_for_low-level_hooks_and_exceptions.patch#L1). Although, we could just edit the patch to handle that instead.

Reply:
Thanks! completely missed that patch.

I am not surprised `BrowserEnvironment` is considered `BROWSER` for the `RUN_AS_NODE` mode. I think the fix here would be to

1) Rename `BrowserEnvironment` enum to `NodeEnvironment`
2) Add a new type `NODE`
3) Set this type when initializing `NodeBindings` from `node_main.cc`

This would avoid mixing `g_standalone_mode` in `ELECTRON_RUN_AS_NODE`. It is better to make the change in this PR.

Reply:
Thanks! completely missed that patch, the fix here would be to

1) Rename `BrowserEnvironment` enum to `NodeEnvironment`
2) Add a new type `NODE`
3) Set this type when initializing `NodeBindings` from `node_main.cc`

This would avoid mixing `g_standalone_mode` in `ELECTRON_RUN_AS_NODE`. It is better to make the change in this PR.

Reply:
Actually wait, since we don't initialize `NodeBindings` in `node_main` , I think simpler fix is to set `g_standalone_mode=false`

Reply:
I thought that'd work as well but it does not seem to.

Reply:
Can you set the variable before `NodeBindings::RegisterBuiltinModules();` call , otherwise its not gonna affect where it needs to.